src/main/java/seedu/address/model/task/TaskDeadline.java

```diff
@@ -26,16 +30,115 @@ public TaskDeadline(String taskDeadline) {
         value = taskDeadline;
     }
 
+
+    /**
+     * Returns the time or date of the task.
+     *
+     * @param line The user's input.
+     * @return The String representation of the time or date of the task.
+     * @throws DateTimeParseException An exception thrown if the date or time is invalid.
+     */
+    public static String parseDateAndTime(Scanner line) {
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("E, MMM dd yyyy");
+        DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("hh:mm a");
+        LocalDate date;
+        LocalTime time;
+        String formattedDateAndTime;
+        String[] split = line.nextLine().split(" ");
+        if (isInDateFormat(split[0])) {
+            date = LocalDate.parse(split[0]);
+            formattedDateAndTime = dateFormatter.format(date);
+            if (split.length == 2) {
+                time = LocalTime.parse(split[1]);
+                formattedDateAndTime = formattedDateAndTime + " " + timeFormatter.format(time);
+            }
+        } else {
+            time = LocalTime.parse(split[0]);
+            formattedDateAndTime = timeFormatter.format(time);
+            if (split.length == 2) {
+                date = LocalDate.parse(split[1]);
+                formattedDateAndTime = formattedDateAndTime + " " + dateFormatter.format(date);
+            }
+        }
+        return formattedDateAndTime;
```

Comment:
so the display can be in the order of both time + date or date + time? Seems a bit inconsistent depending on the order of the input? Would it be better to make it consistent?
![image](https://user-images.githubusercontent.com/69947227/139671088-90a482b7-47a0-4655-8513-1474ab6f4ca2.png)
![image](https://user-images.githubusercontent.com/69947227/139671107-924b99d4-b3c2-490c-9cde-651ee9a154eb.png)

Reply:
Ah yes I made it such that it will be consistent with the order of the user input😬 I can change it such that it always displays the date before time!!

Reply:
Let's make it consistently display the date before the time then? 👍